pkg/util/util.go

```diff
@@ -1,16 +1,102 @@
 package util
 
 import (
+	"errors"
 	"fmt"
+	"io"
+	"io/fs"
 	"os"
 	"path/filepath"
 	"regexp"
 	"time"
 
+	securejoin "github.com/cyphar/filepath-securejoin"
 	"github.com/fsnotify/fsnotify"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
 )
 
+// ConfigureContainerTimeZone configure the time zone for a container.
+// It returns the path of the created /etc/localtime file if needed.
+func ConfigureContainerTimeZone(timezone, containerRunDir, mountPoint, etcPath, mountLabel, containerID string) (string, error) {
+	if timezone == "" {
+		return "", nil
+	}
+	var err error
+	timezonePath := filepath.Join("/usr/share/zoneinfo", timezone)
+	if timezone == "local" {
+		timezonePath, err = filepath.EvalSymlinks("/etc/localtime")
+		if err != nil {
+			return "", fmt.Errorf("finding local timezone for container %s: %w", containerID, err)
+		}
+	}
+
+	etcFd, err := unix.Open(etcPath, unix.O_RDONLY|unix.O_PATH, 0)
```

Comment:
This piece is Linux-specific, so `make build-cross` will fail for other distros: https://github.com/containers/common/pull/1775/checks?check_run_id=19869114459